omnigibson/systems/macro_particle_system.py

```diff
@@ -475,31 +525,48 @@ def generate_group_particles_on_object(cls, group, max_samples, min_samples_for_
         avg_scale = np.cbrt(np.product(obj.scale))
         bbox_extents_global = scales * cls._particle_object.aabb_extent.reshape(1, 3) * avg_scale
 
-        # Sample locations for all particles
-        # TODO: Does simulation need to play at this point in time? Answer: yes
-        results = sample_cuboid_on_object_symmetric_bimodal_distribution(
-            obj=obj,
-            num_samples=max_samples,
-            cuboid_dimensions=bbox_extents_global,
-            bimodal_mean_fraction=cls._SAMPLING_BIMODAL_MEAN_FRACTION,
-            bimodal_stdev_fraction=cls._SAMPLING_BIMODAL_STDEV_FRACTION,
-            axis_probabilities=cls._SAMPLING_AXIS_PROBABILITIES,
-            undo_cuboid_bottom_padding=True,
-            verify_cuboid_empty=False,
-            aabb_offset=cls._SAMPLING_AABB_OFFSET,
-            max_sampling_attempts=cls._SAMPLING_MAX_ATTEMPTS,
-            refuse_downwards=True,
-        )
+        if obj.prim_type == PrimType.CLOTH:
+            # raise NotImplementedError("Sampling for cloth not implemented yet!")
+            # Sample locations based on randomly sampled keyfaces
+            cloth = obj.root_link
+            assert len(cloth.keyface_idx) >= max_samples, \
+                "Insufficient number of keyfaces from which to sample group visual particles for cloth!"
```

Comment:
In this case, maybe we should just sample `len(cloth.keyface_idx)` samples? Even for rigid bodies, `max_samples` is not guaranteed to be satisfied anyways, right?

Reply:
Maybe it's constrained by the number of faces, instead of key faces

Reply:
done!

Reply:
I still see the assert here?
```
assert len(cloth.keyface_idx) >= max_samples
```

Reply:
fixed!